### PR TITLE
DEP: Finalize future warning move in lstsq default

### LINKED ISSUE #11015: linalg.lstsq: Warn that using future default will break for numpy < 1.14
From 1.14, using `linalg.lstsq` without specifying `rcond` yields the following warning:

> `rcond` parameter will change to the default of machine precision times `max(M, N)` where `M` and `N` are the input matrix dimensions.
> To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.

Which seems to strongly indicate that `rcond=None` is the safe bet going further unless you have a reason not to.
However, the warning does not indicate that doing so will make such code break on any install which has still numpy <1.14 (which unfortunately with pip is not so infrequent).
```
TypeError: must be real number, not NoneType
```
This is a minor gripe and proper regression testing on the user end should prevent big problems, but I still think it would be nice to be more diligent when introducing those types of changes.

### PR DESCRIPTION
**Pull request by seberg**
NumPy 2.0 is clearly the (long overdue) time to do this...

Closes gh-11015

### CODE REVIEW

File: doc/release/upcoming_changes/25721.deprecation.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+``np.linalg.lstsq`` now defaults to new ``rcond`` value
+-------------------------------------------------------
+`~numpy.linalg.lstsq` now uses the new rcond value of the machine precision
+times ``max(M, N)``.  Previously, the machine precision was used by a
```
Comment (charris) [inline doc/release/upcoming_changes/25721.deprecation.rst:4]: Some missing words here?

File: numpy/linalg/_linalg.py [hunk 2]
```diff
@@ -2392,13 +2392,12 @@ def lstsq(a, b, rcond="warn"):
...
 
-        .. versionchanged:: 1.14.0
-           If not set, a FutureWarning is given. The previous default
-           of ``-1`` will use the machine precision as `rcond` parameter,
-           the new default will use the machine precision times `max(M, N)`.
-           To silence the warning and use the new default, use ``rcond=None``,
-           to keep using the old behavior, use ``rcond=-1``.
+        .. versionchanged:: 2.0
+            Previously, the default was ``-1``, but a warning was given that
+            this will change.
```
Comment (charris) [inline numpy/linalg/_linalg.py:2400]: will -> would

File: doc/release/upcoming_changes/25721.deprecation.rst (hunk 1 above)
Comment (seberg) [reply doc/release/upcoming_changes/25721.deprecation.rst]: The by was supposed to be a but, I think it's fine then?  (sorry force pushed those two changes)

**Comment by charris** [comment]
Thanks Sebastian.

**Pull request merged**